app/src/main/java/com/bronzeswordstudios/nasaimageviewer/model/MainActivity.kt

```diff
@@ -121,65 +104,34 @@ class MainActivity : AppCompatActivity(), LoaderManager.LoaderCallbacks<ArrayLis
 		connectivityManager.registerDefaultNetworkCallback(object : ConnectivityManager.NetworkCallback() {
 			override fun onAvailable(network: Network) {
 				runOnUiThread(Runnable {
-					if (!isLoaded) {
+					if (imageList.size == 0) {
```

Comment:
This is NOT good. It's an array list. Learn to use this class.

Reply:
Addressed in commit 9322ec1!